File: python/Orchestrations/BlobJob/blob_job_orchestrator.py
import logging
import azure.functions as func
import azure.durable_functions as df


def blob_job_orchestrator(context: df.DurableOrchestrationContext):
    """
    BlobJobOrchestrator - Durable Function의 Orchestrator 함수
    IoTHub_Trigger에 의해 호출됨
    
    Args:
        context: Durable Orchestration Context
    
    Returns:
        Orchestration 결과
    """
    message = context.get_input()
    logging.info(f"BlobJobOrchestrator processed a message: {message}")
    
    # Activity 함수 호출
    result = yield context.call_activity('blob_job_activity1', message)
    
    # Orchestrator 인스턴스 ID 로깅
    logging.info(f"activity result: '{result}'")
    logging.info(f"Orchestration instance ID: {context.instance_id}")
    
    return result


# Orchestrator 함수 빌더
orchestrator = df.Orchestrator.create(blob_job_orchestrator)

File: dotnet/Orchestrations/BlobJob/BlobJobOrchestrator.cs


using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Host;
using Microsoft.Azure.EventHubs;
using System.Text;
using Microsoft.Extensions.Logging;
using Microsoft.Azure.WebJobs.Extensions.DurableTask;
using System.Threading.Tasks;
using Microsoft.Function.Orchestrations.Activities;

namespace Microsoft.Function.Orchestrations.BlobJob
{
    public class BlobJobOrchestrator
    {
        
        //BlobJobOrchestrator 는 Orchestratior 로써 IoTHub_Trigger 가 호출한다.
        [FunctionName(nameof(BlobJobOrchestrator))]
        public async Task Run(
            [OrchestrationTrigger]IDurableOrchestrationContext context,
            ILogger log)
        {
            var message = context.GetInput<string>();
            
            // 보안: Orchestrator에서는 ReplaySafeLogger를 사용해야 하며, 민감한 데이터는 로그에 기록하지 않음
            if (!context.IsReplaying)
            {
                log.LogInformation("BlobJobOrchestrator processing message. InstanceId: {InstanceId}", context.InstanceId);
            }

            //Activity 이름
            string BlobJobActivity = nameof(BlobJobActivity1);

            var result = await context.CallActivityAsync<string>(
                BlobJobActivity,
                message);


            //Orchestrator 인스턴스 ID
            if (!context.IsReplaying)
            {
                log.LogInformation("Activity completed for InstanceId: {InstanceId}", context.InstanceId);
            }

        }
    }
}